File: string/1694_reformat_phone_number/1694_reformat_phone_number.go
package main

import "fmt"

func reformatNumber(number string) string {
	tmp := ""
	for i := 0; i < len(number); i++ {
		if number[i] != ' ' && number[i] != '-' {
			tmp += string(number[i])
		}
	}
	res := ""
	i := 0
	for i = 0; i < len(tmp)-4; i += 3 {
		res += tmp[i:i+3] + "-"
	}
	if i == len(tmp)-4 {
		res += tmp[i:i+2] + "-" + tmp[i+2:]
	} else {
		res += tmp[i:]
	}
	return res
}

func main() {
	number := "1-23-45 6"
	res := reformatNumber(number)
	fmt.Println(res)
}

File: string/1694_reformat_phone_number/1694_reformat_phone_number.py


class Solution(object):
    def reformatNumber(self, number):
        """
        :type number: str
        :rtype: str
        """
        number = number.replace('-' ,'').replace(' ', '')
        ans = []
        for i in range(0, len(number), 3):
            ans.append(number[i:i+3])

        if len(ans) >= 2 and len(ans[-1]) == 1:
            ans[-1] = ans[-2][2] + ans[-1]
            ans[-2] = ans[-2][:2]
        return '-'.join(ans)


number = "1-23-45 6"
res = Solution().reformatNumber(number)
print(res)